src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -73,77 +75,59 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
         } catch (error) {
             const parsedError: IParsedError = parseError(error);
 
-            if (parsedError.errorType !== 'UserCancelledError') {
-                // tslint:disable-next-line: no-floating-promises
-                ext.ui.showWarningMessage(localize('couldNotConnect', 'Could not connect to "{0}": {1}', this.parent.label, parsedError.message));
+            if (parsedError.errorType === invalidCredentialsErrorType || parsedError.errorType === 'CredentialsNotFoundError') {
```

Comment:
The `CredentialsNotFoundError` pattern is a bit odd. I think we get better type safety and a more obvious flow if we do something like this:

```
const { username, password } = await this.getCredentialsFromKeytar();
if (username && password) {
    try {
        // return normal tree items
    } catch (error) {
        if (parsedError.errorType === invalidCredentialsErrorType) {
            // show warning message and fall down below
        } else {
            throw error;
        }
    }
}

// return "Enter Credentials" tree item
```